### DISCUSSION TITLE
`.tsx` files are incorrectly detected as JavaScript instead of TypeScript

### DISCUSSION THREAD

**Discussion by clicktodev** [community/community#147152] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Bug

### Body

.tsx files are typescript files for react components however these files show up when filtering by javascript in project files

example:
https://github.com/search?q=repo%3Afacebook%2Freact-native-website++language%3AJavaScript&type=code

![image](https://github.com/user-attachments/assets/89159e98-fb72-4037-996e-79c484535988)

**Comment by OtayNacef** [comment]
GitHub uses the open source [Linguist library](https://github.com/github/linguist) to determine file languages for syntax highlighting and repository statistics.
I believe it is related to this discussion: [GitHub Linguist #7118](https://github.com/github-linguist/linguist/discussions/7118).

**Comment by look** [comment, marked as answer]
We recently deployed an improved to language detection for content shared across repositories. This file is detected as TSX when I search for it now.

**Reply by clicktodev** [reply]
.tsx files no longer show up in the js filter but they also don't show up in the typescript filter https://github.com/search?q=repo%3Afacebook%2Freact-native-website++language%3ATypeScript&type=code

**Reply by look** [reply]
TSX is considered a separate language ([here's the definition of it](https://github.com/github-linguist/linguist/blob/5293d68060bb1855293fc1ca9c3e184a63716332/lib/linguist/languages.yml#L7317-L7327)).

I see results when using [`language:TSX`](https://github.com/search?q=repo%3Afacebook%2Freact-native-website++language%3ATSX&type=code).

**Reply by clicktodev** [reply]
well in that case jsx should be also be considered a separate language.

also shouldn't tsx show up in the languages summary if it's considered a separate language?

![image](https://github.com/user-attachments/assets/95d5d9c9-765f-4eb2-affb-6f8e805e366e)

**Reply by clicktodev** [reply]
there are jsx files in this js filter https://github.com/search?q=repo%3Afacebook%2Freact-native-website++language%3AJavaScript&type=code

**Reply by clicktodev** [reply]
Also I think jsx and tsx should be part of javascript and typescript because react components can also have the js and ts extensions.

**Reply by look** [reply]
You could propose that in the Linguist project. As it stands, JSX is considered [an alias of JavaScript](https://github.com/github-linguist/linguist/blob/5293d68060bb1855293fc1ca9c3e184a63716332/lib/linguist/languages.yml#L3494) while TSX is its own language.

**Reply by clicktodev** [reply]
> well in that case jsx should be also be considered a separate language.
> 
> also shouldn't tsx show up in the languages summary if it's considered a separate language?
> 
> ![image](https://private-user-images.githubusercontent.com/34947993/401312389-95d5d9c9-765f-4eb2-affb-6f8e805e366e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYzNzI0MDUsIm5iZiI6MTczNjM3MjEwNSwicGF0aCI6Ii8zNDk0Nzk5My80MDEzMTIzODktOTVkNWQ5YzktNzY1Zi00ZWIyLWFmZmItNmY4ZTgwNWUzNjZlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA4VDIxMzUwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ3NTZhZTA1YzZjOWY5YjAyN2I3ZTQyNTQ5OWZkMjAwYWJmMGFlYjEzMDc4NDM2NDg0NzkzMTNmNDk5MzlhMDQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ibXoGRZNqic5VLQovfWXEowpXuM7wckrgpzVIWo6sW0)

alright i'll look into Linguist. And what about ☝️

**Reply by look** [reply]
I am not sure about why it is not shown in the language statistics. That is not generated by the search engine, so the data source is different.

**Reply by clicktodev** [reply]
any idea who to reach about this because tsx files not showing up after clicking typescript from the language statistics is kinda misleading.
